src/components/NewCommentForm.tsx

```diff
@@ -74,26 +115,40 @@ export const NewCommentForm: React.FC = () => {
           <textarea
             id="comment-body"
             name="body"
+            value={newComment?.body}
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={classNames('textarea', { 'is-danger': body })}
+            onChange={(event) => onCommentData(event, newComment)}
```

Comment:
Pass only the 'event' to the onCommentData function. Remove the second argument 'newComment'.